tensorflow/python/ops/clip_ops.py

```diff
@@ -103,7 +104,9 @@ def clip_by_value(t, clip_value_min, clip_value_max,
     values = ops.convert_to_tensor(
         t.values if isinstance(t, indexed_slices.IndexedSlices) else t,
         name="t")
-
+    check_ops.assert_greater_equal_v2(clip_value_max, clip_value_min, message=
```

Comment:
Unfortunately, asserts create synchronization points which can degrade performance significantly, especially in distributed training settings. It might be preferable to make such checks in the kernel implementation itself. 

At L123, there is a TODO to switch to a native kernel. Can we investigate that instead and ensure we make the assertion there?

Reply:
Agreed.